include/shark/Algorithms/QP/BoxConstrainedProblems.h

```diff
@@ -273,6 +273,49 @@ class BoxConstrainedProblem{
 	void reshrink(){}
 	void unshrink(){}
 
+	/// \brief Define the initial solution for the iterative solver.
+	///
+	/// This method can be used to warm-start the solver. It requires a
+	/// feasible solution (alpha) and the corresponding gradient of the
+	/// dual objective function.
+	void setInitialSolution(RealVector const& alpha, RealVector const& gradient)
+	{
+		std::size_t n = dimensions();
+		SIZE_CHECK(alpha.size() == n);
+		SIZE_CHECK(gradient.size() == n);
+		for (std::size_t i=0; i<n; i++)
+		{
+			SHARK_ASSERT(alpha(i) >= boxMin(i) && alpha(i) <= boxMax(i));
+			std::size_t j = permutation(i);
+			m_problem.alpha(i) = alpha(j);
+			m_gradient(i) = gradient(j);
+			m_alphaStatus[i] = ((alpha(i) == boxMin(i)) ? AlphaLowerBound : 0)
+			                 | ((alpha(i) == boxMax(i)) ? AlphaUpperBound : 0);
+		}
+	}
+
+	/// \brief Define the initial solution for the iterative solver.
+	///
+	/// This method can be used to warm-start the solver. It requires a
+	/// feasible solution (alpha), for which it computes the gradient of
+	/// the dual objective function. Note that this is a quadratic time
+	/// operation in the number of non-zero coefficients.
+	void setInitialSolution(RealVector const& alpha)
+	{
+		std::size_t n = dimensions();
+		SIZE_CHECK(alpha.size() == n);
+		RealVector gradient = m_problem.linear;
+		std::vector<QpFloatType> q(n);
```

Comment:
Hmmm... but what's QpFloatType then good for? I could do blas::vector<QpFloatType> though.